### PR TITLE
make the dataset splits enums

### PR DESCRIPTION
**Pull request by olliethomas**

### CODE REVIEW

File: ethicml/data/tabular_data/admissions.py [hunk 1]
```diff
@@ -37,19 +37,31 @@
...
 __all__ = ["admissions"]
 
-AdmissionsSplits = Literal["Gender"]
 
+class AdmissionsSplits(Enum):
+    GENDER = "Gender"
+    CUSTOM = "Custom"
+
+
+VALID_STRS = Literal[tuple([e.value for e in AdmissionsSplits])]  # type: ignore[misc]
```
Comment (thomkeh) [inline ethicml/data/tabular_data/admissions.py:55]: You may as well use `str`. No static type checker can parse this (I think?).

File: ethicml/data/tabular_data/admissions.py [hunk 2]
```diff
@@ -37,19 +37,31 @@
...
 
 from ..dataset import Dataset
 from ..util import flatten_dict
 
 __all__ = ["admissions"]
 
-AdmissionsSplits = Literal["Gender"]
 
+class AdmissionsSplits(Enum):
+    GENDER = "Gender"
```
Comment (thomkeh) [inline ethicml/data/tabular_data/admissions.py:51]: All caps for the values?

**Review by thomkeh — APPROVED** [review]

**Comment by olliethomas** [comment]
The motivation is that when I call for example the `adult` function and pass `Race`, I get "expected Literal not str".

File: ethicml/data/tabular_data/admissions.py (hunk 2 above)
Comment (olliethomas) [reply ethicml/data/tabular_data/admissions.py:51]: Apparently all caps for the name is the done thing according to pylint. I'm not sure I like it

File: ethicml/data/tabular_data/admissions.py (hunk 1 above)
Comment (olliethomas) [reply ethicml/data/tabular_data/admissions.py:55]: Yeah...

**Pull request merged**